src/item/Elytra.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\entity\Living;
+use pocketmine\event\entity\EntityDamageEvent;
+use pocketmine\math\Vector2;
+
+class Elytra extends Armor{
+
+	private const APPLY_DAMAGE_INTERVAL = 20;
+	private int $applyDamageTime = self::APPLY_DAMAGE_INTERVAL;
+
+	public function onTickWorn(Living $entity) : bool {
+		if ($entity->isGliding()) {
+
+			$direction = $entity->getDirectionVector();
+			$yVel = $direction->y;
+
+			if($yVel < 0.5){
```

Comment:
That isn't how you get the velocity. It's supposed to be `currentY - prevY` but there needs to be a way to get the previous position outside of the `Entity` class

Reply:
> It's supposed to be currentY - prevY

Yes, I know but there is no method for it - at least nothing found... Is there anything outside the entity class or you didn´t know?

I could log the previous Position but idk if this is a great idea

Reply:
private $previousPosition = null;

    public function onTickWorn(Living $entity): bool {
        $currentPosition = $entity->getLocation();

        if ($this->previousPosition === null) {
            // Store the current position as the previous position in the first tick
            $this->previousPosition = $currentPosition;
        }

        $deltaY = $currentPosition->getY() - $this->previousPosition->getY();
        $deltaXY = $currentPosition->distance($this->previousPosition);

        // Calculate vertical and horizontal velocities
        $yVelocity = $deltaY;
        $xyVelocity = $deltaXY;

        if ($yVelocity > $xyVelocity) {
        }

        // Update the previous position for the next tick
        $this->previousPosition = $currentPosition;

        return false;
    }

Reply:
> ue about it and wait until that method gets added or make a PR that adds it.

would it be bad if i first add myself what i sent above and later add the new method?